### DISCUSSION TITLE
[@types/jscodeshift] missing exportKind in [namedTypes.ExportNamedDeclaration]

### DISCUSSION THREAD

**Discussion by Marco-Daniel** [DefinitelyTyped/DefinitelyTyped#69615] [Issues with a @types package]
I've been using jscodeshift to transform export statements. If I don't set the `exportKind` key on my created exportDeclaration made using `j.exportNamedDeclaration` function then it wont't work properly, so it does work, it just seems the types are missing.

![image](https://github.com/DefinitelyTyped/DefinitelyTyped/assets/48966669/11b062d3-6238-455d-82a1-85cf08df8314)

I think it should be the same as `ImportDeclaration.importKind`

[a comment by an automated account is not shown]

**Comment by sztobar** [comment]
I haven't touched jscodeshift in a while, could you provide some small code example and share it using [https://astexplorer.net](https://astexplorer.net)?

**Comment by Marco-Daniel** [comment]
@sztobar Not sure if I understand you correctly, but what i'm trying to modify are export statements like:
https://astexplorer.net/#/gist/cc96edd3f2b8fe10c5665c681e0dc70e/cfd24ce0a658c6e6b1c44cc90e2ce7d806cbce49

In my own code i have the following right now:
```
const exportDeclaration = j.exportNamedDeclaration(
        astPath.node.declaration,
        astPath.node.specifiers,
        j.literal(transformImportPath({ filePath, dir, importPath: astPath.node.source.value.toString() })),
      );

      // @ts-expect-error the types from jscodeshift are not right, this prop is needed to keep the export how it should
      // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
      exportDeclaration.exportKind = astPath.node.exportKind;

      j(astPath).replaceWith(exportDeclaration);
```

If I dont add the line `exportDeclaration.exportKind = astPath.node.exportKind;` it removes the type keyword from the export, only then I get typescript errors. Hope this makes sense?
